projects/dotcms-field-elements/src/components/dot-binary-file-preview/dot-binary-file-preview.tsx

```diff
@@ -0,0 +1,63 @@
+import { Component, Element, Event, EventEmitter, Prop } from '@stencil/core';
+import Fragment from 'stencil-fragment';
+import { DotCMSTempFile } from '../../models';
+
+/**
+ * Represent a dotcms text field for the binary file preview.
+ *
+ * @export
+ * @class DotBinaryFilePreviewComponent
+ */
+@Component({
+    tag: 'dot-binary-file-preview',
+    styleUrl: 'dot-binary-file-preview.scss'
+})
+export class DotBinaryFilePreviewComponent {
+    @Element() el: HTMLElement;
+
+    /** (optional) file to be displayed */
+    @Prop({ reflectToAttr: true, mutable: true })
+    file: DotCMSTempFile = null;
+
+    /** (optional) Delete button's label */
+    @Prop({ reflectToAttr: true })
+    deleteLabel = 'Delete';
+
+    /** Emit when the file is deleted */
+    @Event() delete: EventEmitter;
+
+    render() {
+        return this.file ? (
+            <Fragment>
+                {this.getPreviewElement()}
+                <div class="dot-file-preview__info">
+                    <span class="dot-file-preview__name">
+                        {this.file ? this.file.fileName : ''}
+                    </span>
+                    <button onClick={() => this.clearFile()}>{this.deleteLabel}</button>
+                </div>
+            </Fragment>
+        ) : null;
+    }
+
+    private clearFile(): void {
+        this.delete.emit(this.file);
+        this.file = null;
+    }
+
+    private getPreviewElement(): JSX.Element {
```

Comment:
Shouldn't this be their own internal component?

Reply:
to small for that.